src/frontend/src/service/cookie-service.ts

```diff
@@ -0,0 +1,35 @@
+interface Cookie {
+  auth_token: string;
+  user_id: string;
+}
+
+export function decodeCookie() {
+  // this decodes the cookie creating an object with {key: `value`} pairs
+  // https://gist.github.com/rendro/525bbbf85e84fa9042c2#gistcomment-2784930
+  const data: any = document.cookie.split(';').reduce((res, c) => {
+    const [key, val] = c
+      .trim()
+      .split('=')
+      .map(decodeURIComponent);
+    const allNumbers = (str: string) => /^\d+$/.test(str);
+    try {
+      return Object.assign(res, {
+        [key]: allNumbers(val) ? val : JSON.parse(val)
```

Comment:
I've taken out the 'allNumbers' realizing that it isn't something that we really need to check for in the basis of our app.

I have been having trouble attempting to get the format of:
```
{
  auth_token: ...,
  user_id: ...
}
```

I am able to type the cookie inputs like so:

```
interface Cookie {
  [key: string]: string;
};
```

when trying to give the `key` a value such as an `enum` or `type` of it's own it starts to give errors about conflicting types and suggests using "object mapping" instead(not entirely sure what to do here).

Reply:
https://stackoverflow.com/questions/13315131/enforcing-the-type-of-the-indexed-members-of-a-typescript-object <- not sure if this helps.

If Typescript's causing problems, just use regular strings for the object keys, or use the any type.

Or just write something like `if (key === 'auth_token')` which is fine too... the declarative programming approach was just an idea in case you were hoping to write something more general, definitely not mandatory to take that approach.

Reply:
That's actually one of the articles I used that brought me to this solution

`interface StringMap { [s: string]: string; }`
vs:
`interface Cookie { [key: string]: string; };`